tests/cases/conformance/types/mapped/mappedTypeConstraints2.ts

```diff
@@ -57,5 +83,5 @@ type ObjectWithUnderscoredKeys<K extends string> = {
 };
 
 function genericTest<K extends string>(objectWithUnderscoredKeys: ObjectWithUnderscoredKeys<K>, key: K) {
-  const shouldBeTrue: true = objectWithUnderscoredKeys[`_${key}`];
+  const shouldBeTrue: true = objectWithUnderscoredKeys[`_${key}`]; // assignability fails here, but ideally should not

```

Comment:
Those kinds of cases could always be OK if we could determine that the template type doesn't depend on the original `K`. I don't think the compiler is equipped to check this sort of thing right now though.